### PR TITLE
AWconditioning

### PR DESCRIPTION
**Pull request by FAznaran**
Small changes to the FInAT file for the Arnold--Winther elements, to make the transformation of the internal DOFs consistent with the zany theory, and to post-hoc rescale the DOFs so as to give the mass matrices O(1) conditioning.

### CODE REVIEW

File: finat/aw.py [hunk 1]
```diff
@@ -3,7 +3,9 @@
 
 import FIAT
 
-from gem import Literal, ListTensor
+#from gem import Literal, ListTensor
```
Comment (rckirby) [inline finat/aw.py:6]: Remove the commmented line

File: finat/aw.py [hunk 2]
```diff
@@ -27,23 +29,39 @@ def _edge_transform(T, coordinate_mapping):
     J_np = numpy.array([[J[0, 0], J[0, 1]],
                         [J[1, 0], J[1, 1]]])
     JTJ = J_np.T @ J_np
-
+
```
Comment (rckirby) [inline finat/aw.py:32]: flake (white space)

File: finat/aw.py [hunk 3]
```diff
@@ -27,23 +29,39 @@ def _edge_transform(T, coordinate_mapping):
     J_np = numpy.array([[J[0, 0], J[0, 1]],
```
Comment (rckirby) [inline finat/aw.py:29]: Is there syntactic sugar in gem to make this cleaner (so you don't have to explicitly index/wrap)?

**Review by rckirby — COMMENTED** [review]
I marked a couple of obvious flake things, and you should fix the rest of the ones that are failing.

**Review by rckirby — CHANGES_REQUESTED** [review]
Also, do we want to include a test confirming an O(1) condition number?

**Comment by wence-** [comment]
@FAznaran is it possible to add a test of the conditioning? I suppose the idea is that the mass matrix should have O(1) conditioning on the reference cell? I think one can write a finat-only test to build the mass matrix and then get numeric values out using the gem evaluator.

**Comment by FAznaran** [comment]
What does @pefarrell think? He's said before that if a test depends on Firedrake, it should live there. IIUC [my PR](https://github.com/firedrakeproject/firedrake/pull/2241) is currently failing its own test, I think simply since _this_ PR is not merged, which is an argument for making the test finat-only.

**Comment by rckirby** [comment]
The test I am suggesting could be done entirely in FInAT (+FIAT+numpy)
without loading any Firedrake meshes, solvers, etc.  But they have the
distinct disadvantage of not existing yet.
The Firedrake PR with Firedrake-level tests couldn't be put in as a FInAT
test and would need to stay in Firedrake.

**Comment by pefarrell** [comment]
My opinion is if the (i) FInAT-only test doesn't yet exist and (ii) it's quick to run in the firedrake test suite, then we should just keep things as-is with these PRs.

**Pull request merged**